openfl/utilities/dataloading.py

```diff
@@ -1,72 +1,87 @@
 # Copyright 2020-2024 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
+import importlib
+import logging
 import os
 import zipfile
-from typing import Union
 
 from openfl.federated import Plan
 from openfl.federated.data.loader import DataLoader
-from openfl.utilities.mocks import MockDataLoader
 
+logger = logging.getLogger(__name__)
 
-def get_dataloader(
-    plan: Plan,
-    prefer_minimal: bool = False,
-    input_shape: Union[list, dict] = None,
-    collaborator_index: int = 0,
-) -> DataLoader:
-    """Get dataloader instance from plan
 
-    NOTE: if `prefer_minimal` is False, cwd must be the workspace directory
-    because we need to construct dataloader from actual collaborator data path
-    with actual data present.
+def initialize_minimal_dataloader(plan: Plan) -> DataLoader:
+    """Initialize a minimal dataloader without loading actual data.
+
+    This is used primarily for model initialization when the actual data
+    is not needed.
 
     Args:
-        plan (Plan):
-            plan object linked with the dataloader
-        prefer_minimal (bool ?):
-            prefer to use MockDataLoader which can be used to more easily
-            instantiate task_runner without any initial data.
-            Default to `False`.
-        input_shape (list | dict ?):
-            input_shape denoted by list notation `[a,b,c, ...]` or in case
-            of multihead models, dict object with individual layer keys such
-            as `{"input_0": [a,b,...], "output_1": [x,y,z, ...]}`
-            Defaults to `None`.
-        collaborator_index (int ?):
-            which collaborator should be used for initializing dataloader
-            among collaborators specified in plan/data.yaml.
-            Defaults to `0`.
+        plan (Plan): Plan object linked with the dataloader
 
     Returns:
-        data_loader (DataLoader): DataLoader instance
+        DataLoader: A minimal dataloader instance with no data loaded
+
+    Raises:
+        ValueError: If required configuration is missing or dataloader class cannot be found
     """
+    try:
```

Comment:
Can we rewrite this block without nested (and ideally without) `try...except` blocks?